src/main/java/org/opentripplanner/raptor/rangeraptor/multicriteria/McStopArrivals.java

```diff
@@ -58,21 +64,21 @@ public int bestArrivalTime(int stopIndex) {
   public boolean reachedByTransit(int stopIndex) {
     return (
       arrivals[stopIndex] != null &&
-      arrivals[stopIndex].stream().anyMatch(ArrivalView::arrivedByTransit)
+      arrivals[stopIndex].stream().anyMatch(McStopArrivals::arrivedByTransit)
```

Comment:
Deferring to an intermediate method does not increases readability. Could use an lambda expression directly instead. Smack on a static import and it's even shorter than the method reference.

Reply:
I don't think I quite understand what you are requesting. Do you mean something like this? https://github.com/opentripplanner/OpenTripPlanner/commit/6a773b0c8e63bb5331abae7168fe34c32467db3a

Reply:
Yes, exactly. Although I would have used 'a' as parameter name.

Reply:
IntelliJ sure does prefer method references, and I guess the belief is that `x -> x...` is a slightly messier syntax. However, a method reference doesn't work if you need to pass a parameter as well, and in that case I think the lambda expression syntax is perfectly fine to use. This avoids a method implementation and having to read another call stack level down to understand the code. Less is better.

I think static imports could and should be used in cases where the values are well known and it adds to readability. The introduction of the `var` keyword into the language surely added weight behind this argument, since it requires IDE support to read out the actual type (something that currently is lacking in for example Github).

I think the aversion to static imports was formulated in a time where not even enums was around, and there was a concern that developers would static-import methods from other classes and create an unreadable spaghetti mess. We now have other ways of doing that. :slightly_smiling_face: